inc/admin.php

```diff
@@ -2069,4 +2074,53 @@ public function survey_category( $value, $scale = 1 ) {
 
 		return 0;
 	}
+
+	/**
+	 * Determines whether the exceed quota warning should be displayed to users.
+	 *
+	 * This function checks if the user's quota usage has exceeded a predefined limit
+	 * and returns a boolean value indicating whether the warning should be shown.
+	 *
+	 * @return bool True if the exceed quota warning should be displayed, false otherwise.
+	 */
+	public function should_show_exceed_quota_warning() {
+		$current_screen = get_current_screen();
+		if ( ( defined( 'DOING_AJAX' ) && DOING_AJAX ) ||
+			is_network_admin() ||
+			! current_user_can( 'manage_options' ) ||
+			! $this->settings->is_connected() ||
+			empty( $current_screen )
+		) {
+			return false;
+		}
+		if ( get_option( 'optml_notice_hide_upg', 'no' ) === 'yes' ) {
+			return false;
+		}
+
+		if ( ! str_contains( $current_screen->base, 'page_optimole' ) ) {
+			return false;
+		}
+		$service_data = $this->settings->get( 'service_data' );
+
+		if ( ! isset( $service_data['plan'] ) ) {
+			return false;
+		}
+		if ( $service_data['plan'] !== 'free' ) {
+			return false;
+		}
+
+		if ( $service_data['days_since_registration'] <= 14 ) {
+			return false;
+		}
+		$visitors_limit     = isset( $service_data['visitors_limit'] ) ? (int) $service_data['visitors_limit'] : 0;
+		$visitors_left      = isset( $service_data['visitors_left'] ) ? (int) $service_data['visitors_left'] : 0;
+		$used_quota         = $visitors_limit - $visitors_left;
+		$is_50_percent_used = ( $used_quota / $visitors_limit ) >= 0.5;
```

Comment:
I think by default we should use something different from 0 for the visitor limit fallback. Otherwise we're dividing by 0 which would throw a fatal error.

I think we can simply chain all the service vars in an isset call and just return false if they're not set.

Reply:
I've added condition to return false